create-image.sh

```diff
@@ -43,6 +43,19 @@ if [ ! "$(ls -A $JETSON_BUILD_DIR)" ]; then
         printf "\e[32mDownload L4T...       "
         wget -qO- $BSP | tar -jxpf - -C $JETSON_BUILD_DIR
         printf "[OK]\n"
+
+        # Fix nvidia's bugs in various BSP scripts
+        case "$BSP" in
+            *32.5*)
+                # Fix link dereferencing in 32.5 for xavier
+                sed -i 's/cp -f/cp -af/g' "$JETSON_BUILD_DIR/Linux_for_Tegra/tools/ota_tools/version_upgrade/ota_make_recovery_img_dtb.sh"
+                ;;
+            *32.6*)
+                # Our image is so small that 10% extra disk space is not enough so we add at least 128 on top of the 10%
+                sed -i 's/rootfs_size +/rootfs_size + 128 +/g' "/tmp/jetson-builder/build/Linux_for_Tegra/tools/jetson-disk-image-creator.sh"
```

Comment:
Is 128 bytes here? Or MB?

It would be good to elaborate in the comment about the magnitude of this number and the origin of 10%.

I know that on first-boot-up I have a script that runs to resize the root partition to fill any remaining space, so this is a step that happens before that I imagine?

Reply:
It's 128MiB.
This happens during nvidia's original flash.sh part of the process, when preallocating file for root partition.
That 128MiB is a completely arbitrary number. I tried increasing it in smaller increments, but even 60MiB extra was not enough in my case. This should really be replaced with some elaborate way of calculating the actual required disk space based on the difference in filesystems on the host and in the root partition.

1. Original script uses `du -ms $ROOTFS_DIR` to determine the required size of the root image.
2. Then it preallocates a file for the root partition with ($that_size + ($that_size/10)). There's a comment in that script saying they are doing the +10% because partition needs to be slightly bigger to accommodate extra space used by inodes and other metadata. This is very unreliable, and probably the reason why I'm hitting this edge case.
3. Then it creates a loopback device with that file and copies the content of $ROOTFS_DIR to it

The problem I've had with rootfs that's ~400MiB big is that the script failed when it tried to copy stuff from rootfs directory to the root partition, and reported that APPSIZE is too small. Note that this is all original nvidia stuff with 0 modifications from your or me.